### PR TITLE
Swap `a.Variant` and `a.Flavor` to match new artifact naming scheme

### PR DESCRIPTION
**Pull request by mevatron**
I ran into an issue with the variant and flavor fields being swapped in the new v2.4.2 artifacts, so I created the tiniest of PRs to fix it :smile: If the naming change wasn't intended, feel free to close this PR! I tested these changes locally and they seem to work at least for the ubuntu-20.04 flavor I was testing out.

### CODE REVIEW

**Comment by mauromorales** [comment]
Hi @mevatron thanks a lot for the PR, you were right on spot, we changed the artifact name again so this was needed. I just took the freedom to extend your PR because we want to handle the artifact names in the 3 different cases (pre 2.4.0, between 2.4.0 and 2.4.1, and 2.4.2 and beyond).

File: pkg/schema/artifact.go [hunk 1]
```diff
@@ -39,23 +39,19 @@ func (a ReleaseArtifact) FileName() string {
 
 	}
 
-	var re = regexp.MustCompile(`v(?P<major>\d+)\.(?P<minor>\d+).+`)
-	match := re.FindStringSubmatch(a.ReleaseVersion)
-	result := make(map[string]string)
-	for i, name := range re.SubexpNames() {
-		if i != 0 && name != "" {
-			result[name] = match[i]
+	if semver.Compare(a.ReleaseVersion, "v2.4.0") < 0 {
```
Comment (Itxaka) [inline pkg/schema/artifact.go:42]: YEEES

Comment (Itxaka) [reply pkg/schema/artifact.go:42]: I did not know we were doing regex for the versions xD

**Review by Itxaka — APPROVED** [review]
looks good, awesome with the test. Was gonna ask if this worked with the k3s artifacts correctly but tests indicate that they do.

Maybe expand the docs to show how to use the k3s stuff is the only thing Im missing?

File: README.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ Check out the full [reference of AuroraBoot  in our documentation](https://kairo
 For instance, in one machine from your workstation, you can run:
 
 ```bash
-$ docker run --rm -ti --net host quay.io/kairos/auroraboot --set "artifact_version=v1.5.0" --set "release_version=v1.5.0" --set "flavor=rockylinux" --set repository="kairos-io/kairos" --cloud-config /....
+$ docker run --rm -ti --net host quay.io/kairos/auroraboot --set "artifact_version=v2.4.2" --set "release_version=v2.4.2" --set "flavor=rockylinux"--set "flavor_release=9"  --set repository="kairos-io/kairos" --cloud-config /....
```
Comment (jimmykarily) [inline README.md:77]: shouldn't this be "flavor_version" like below?

Comment (jimmykarily) [reply README.md:77]: or even better "flavor_release" everywhere so that we are consistent with Kairos repo?

Comment (mauromorales) [reply README.md:77]: shoot, good catch

File: pkg/schema/artifact.go (hunk 1 above)
Comment (mauromorales) [reply pkg/schema/artifact.go:42]: yup that was a stupid decision from me from the past 🤣

**Pull request merged**